### DISCUSSION TITLE
AssemblyFixture sample

### DISCUSSION THREAD

**Discussion by macfarmw** [xunit/xunit#2509] [Question] [answered]
I setup the [AssemblyFixture sample](https://github.com/xunit/samples.xunit/tree/main/AssemblyFixtureExample) in my test project. It works great! I now need to initialize a resource that must be called with async. I attempted to add the IAsyncLifetime to my assembly fixture class, but the InitializeAsync method does not fire.  

How can I modify the sample so that it will work with IAsyncLifetime? If that's not possible, is there an effective way to run the async initialize code in the constructor? I see GetAwaiter().GetResult() used frequently, but I think that leads to other problems and I'd like to avoid it. IAsyncLifetime seems ideal if I can get wired up correctly.

**Comment by bradwilson** [comment, marked as answer]
I don't think `IAsyncLifetime` is going to work here in v2, so I would resort to `.GetAwaiter().GetResult()`.

This is working w/ `IAsyncLifetime` in v3.
